drivers/tty/serial/serial_core.c

```diff
@@ -3404,6 +3405,22 @@ int serial_core_register_port(struct uart_driver *drv, struct uart_port *port)
 	 */
 	port->flags |= UPF_DEAD;
 
+	if (port->dev == NULL) {
+		static int next_port_number = 0;
```

Comment:
nitpick: `static unsigned int`

Reply:
If you change this be sure to change the format string to %u.

I don't really think it matters either way. 2^31 and 2^32 possibilities should both be plenty.

Reply:
Also checkpatch complains about:
```
ERROR: do not initialise statics to 0
#31: FILE: drivers/tty/serial/serial_core.c:3409:
+               static int next_port_number = 0;
```
That's because static variables already default to zero and is more efficient if you leave the initialization out (the compiler can place them in the .bss segment which gets zero-out in one go). You can safely remove the '=0' initialization.

Reply:
Updated to `static unsigned int`